internal/provider/client.go

```diff
@@ -2,123 +2,73 @@ package provider
 
 import (
 	"crypto/tls"
-	"encoding/json"
-	"fmt"
 	"io"
 	"net/http"
+	"net/url"
 	"strings"
+	"time"
 )
 
-// Client -
-type AAPClient struct {
-	HostURL            string
-	Username           *string
-	Password           *string
-	InsecureSkipVerify bool
-}
-
-// ansible host
-type AnsibleHost struct {
-	Name      string            `json:"name"`
-	Groups    []string          `json:"groups"`
-	Variables map[string]string `json:"variables"`
+// Provider Http Client interface (will be useful for unit tests)
+type ProviderHttpClient interface {
+	doRequest(method string, path string, data io.Reader) (int, []byte, error)
 }
 
-// ansible host list
-type AnsibleHostList struct {
-	Hosts []AnsibleHost `json:"hosts"`
+// Client -
+type AAPClient struct {
+	HostURL    string
+	Username   *string
+	Password   *string
+	httpClient *http.Client
 }
 
-// NewClient -
-func NewClient(host string, username *string, password *string, insecure_skip_verify bool) (*AAPClient, error) {
+// NewClient - create new AAPClient instance
+func NewClient(host string, username *string, password *string, insecure_skip_verify bool, timeout int64) (*AAPClient, error) {
+	hostURL := host
+	if !strings.HasSuffix(hostURL, "/") {
+		hostURL = hostURL + "/"
+	}
 	client := AAPClient{
-		HostURL:            host,
-		Username:           username,
-		Password:           password,
-		InsecureSkipVerify: insecure_skip_verify,
+		HostURL:  hostURL,
+		Username: username,
+		Password: password,
+	}
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: insecure_skip_verify},
 	}
+	// Initialize http client using default timeout = 5sec
```

Comment:
Where is the default timeout set to 5? It looks like the provider just uses a 0 value if nothing is passed?

Reply:
updated